NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -129,9 +136,21 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
         cell = bigCardView;
     }
     
+    CCBRNewsCardViewModel *itemViewModel = [self.viewModel itemViewModelAtIndex:indexPath.row];
+    [[CCBREventLogger shared]logCardImpression:[itemViewModel newsFeedId]];
+
     return cell;
 }
 
+
+-(void)collectionView:(UICollectionView *)collectionView willDisplayCell:(UICollectionViewCell *)cell forItemAtIndexPath:(NSIndexPath *)indexPath
+{
+    if((indexPath.row) == self.viewModel.itemCount-1)
+    {
+        [self.viewModel loadMoreItem];
+    }
```

Comment:
Because View/ViewController should be as dumb as possible, these logic should be moved to ViewModel.